grpc-gcp/src/main/java/com/google/cloud/grpc/GcpManagedChannel.java

```diff
@@ -1084,15 +1086,45 @@ private synchronized ChannelRef createNewChannel() {
     return channelRef;
   }
 
+  // Returns first newly created channel or null if there are already some channels in the pool.
+  @Nullable
+  private ChannelRef createFirstChannel() {
```

Comment:
Why you want to know if one channel is the first channel? Otherwise this method can be replaced by `tryCreateNewChannel()`.

Reply:
This method is extracted from `pickLeastBusyChannel` L1125 below.
This is for the case when several RPCs are initiated concurrently when channel pool is empty. This is why we need to check `if (channelRefs.isEmpty())` after synchronization -- not to create unnecessary channels. E.g., 5 RPCs initiated but all of them could use a single channel because lowwatermark is set higher than 5, this method will create only 1 channel while `tryCreateNewChannel` will create upto 5 or maxSize.